contrib/scripts/create_nbtc_proof.py

```diff
@@ -0,0 +1,157 @@
+
+    # Generate the proof by traversing up the tree
+    while len(all_hashes) > 1:
+        # If the index is even, get the next hash as the sibling
+        if index % 2 == 0:
+            if index + 1 < len(all_hashes):
+                proof.append(all_hashes[index + 1])
+            else:
+                proof.append(all_hashes[index])  # If odd number, duplicate the last
+                # If the index is odd, get the previous hash as the sibling
+        else:
+            proof.append(all_hashes[index - 1])
+        if len(all_hashes) % 2 != 0:
+            all_hashes.append(all_hashes[-1])
+
+        # Pair up the hashes and hash them again
+        all_hashes = [
+            double_sha256(all_hashes[i] + all_hashes[i + 1])
+            for i in range(0, len(all_hashes), 2)
+        ]
+        index = index // 2  # Move up one level in the tree
+
+    return proof
+
+
+# Example transaction hashes (replace with actual transaction hashes)
+
+
+def big_endian_to_little_endian(hex_str):
+    """Converts a big-endian hex string to little-endian."""
+    # Ensure the hex string has an even length, as each byte is 2 characters
+    if len(hex_str) % 2 != 0:
+        raise ValueError("Hex string must have an even length.")
+
+    # Reverse the order of bytes (pair of hex characters)
+    little_endian = "".join(
+        [hex_str[i : i + 2] for i in range(0, len(hex_str), 2)][::-1]
+    )
+
+    return little_endian
+
+
+def read_byte_transaction(hex_str):
+    tx_data = {
+        "version": "",
+        "input_count": 0,
+        "inputs": "",
+        "output_count": 0,
+        "outputs": "",
+        "lock_time": ""
+    }
+
+    i = 4;
+    tx_data["version"] = hex_str[:i * 2]
+
+    if (tx_data["version"] == "02000000"):
+        i += 2
+    
+    tx_data["input_count"] = int(hex_str[i * 2: (i + 1) * 2], 16)
+    i += 1
+    for j in range(tx_data["input_count"]):
+        tx_data["inputs"] = tx_data["inputs"] + hex_str[i * 2 : (i + 32 + 4) * 2]
+        i += 32 + 4
+        
+        k = int(hex_str[i * 2: (i + 1) * 2], 16)
+        tx_data["inputs"] = tx_data["inputs"] + hex_str[i * 2: (i + 1) * 2]                
+        i += 1
+        tx_data["inputs"] = tx_data["inputs"] + hex_str[i * 2 : (i + k + 4) * 2]
+        i += k + 4
+
+    tx_data["output_count"] = int(hex_str[i * 2: (i + 1) * 2], 16)
+    i += 1;
+
+    for j in range(tx_data["output_count"]):
+        tx_data["outputs"] = tx_data["outputs"] + hex_str[i * 2 : (i + 8) * 2]
+        i += 8
+        k = int(hex_str[i * 2: (i + 1) * 2], 16)
+        tx_data["outputs"] = tx_data["outputs"] + hex_str[i * 2: (i + 1) * 2]
+        i += 1
+        tx_data["outputs"] = tx_data["outputs"] + hex_str[i * 2 : (i + k) * 2]
+        i += k
+    tx_data["lock_time"] = hex_str[-8:]
+    
+    return tx_data
+    
+def main():
+    parser = argparse.ArgumentParser(description="nBTC prepare data")
+    parser.add_argument('block_hash', type=str, help='block hash')
```

Comment:
should be block filename
```suggestion
    parser.add_argument('block_filename', type=str, help='path to a JSON file with Bitcoin block data (can be created using "contrib/scripts/create_btc_mint_data.sh"')
```